From 621cb2c78b253ee023a86bc91d6ddb048481c04c Mon Sep 17 00:00:00 2001
From: Sebastian Pipping <sebastian@pipping.org>
Date: Tue, 27 Oct 2026 17:57:19 +0100
Subject: [PATCH] lib: Protect `nPrefixes` in `storeAtts` from signed integer
 overflow

.. by migrating to `size_t`
[Hades: from libexpat pull request #1450, commit 621cb2c78b253ee023a86bc91d6ddb048481c04c,
 not merged as of 2026-10-28; taken for CVE-2026-117610 and CVE-2026-118370.]

---
 lib/xmlparse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/xmlparse.c b/lib/xmlparse.c
index a04ffde..84a1fc1 100644
--- a/lib/xmlparse.c
+++ b/lib/xmlparse.c
@@ -4170,7 +4170,7 @@ storeAtts(XML_Parser parser, const ENCODING *enc, const char *attStr,
   DTD *const dtd = parser->m_dtd; /* save one level of indirection */
   int attIndex = 0;
   XML_Char *uri;
-  int nPrefixes = 0;
+  size_t nPrefixes = 0;
   BINDING *binding;
   const XML_Char *localPart;
 
-- 
2.53.0

